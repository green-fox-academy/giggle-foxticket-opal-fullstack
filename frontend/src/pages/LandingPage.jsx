import React from 'react';
import MainFeatures from '../components/MainFeatures/MainFeatures';

const LandingPage = () => {
<<<<<<< HEAD
  return <></>;
=======
  return (
    <>
      I am the landing page!
      <MainFeatures />
    </>
  );
>>>>>>> master
};

export default LandingPage;
