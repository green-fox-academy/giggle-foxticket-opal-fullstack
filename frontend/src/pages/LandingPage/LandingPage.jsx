import React from 'react';
import { Link } from 'react-router-dom';
import MainFeatures from '../../components/MainFeatures/MainFeatures';
import ImageComposition from '../../components/ImageComposition/ImageComposition';
import Footer from '../../components/Footer/Footer';
import VideoPlayer from '../../components/VideoPlayer/VideoPlayer';
import Button from '../../components/Button/Button';
import './LandingPage.styles.sass';
import SubscribeMail from '../../components/SubscribeMail/SubscribeMail';

const LandingPage = () => {
  return (
    <>
      <ImageComposition />
      <MainFeatures />
      <Link to="/shop">
        <Button buttonStyle="btn--danger--solid">Buy Tickets</Button>
      </Link>
      <VideoPlayer />
      <SubscribeMail />
      <Footer />
    </>
  );
};

export default LandingPage;
