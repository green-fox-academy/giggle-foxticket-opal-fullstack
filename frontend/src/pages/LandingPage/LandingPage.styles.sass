.btn--danger--solid
  position: relative
  left: 50%
  transform: translateX(-50%)
  margin-bottom: 45px
