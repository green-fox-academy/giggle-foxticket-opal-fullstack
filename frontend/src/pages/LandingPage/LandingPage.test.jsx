import React from 'react';
import renderer from 'react-test-renderer';
import LandingPage from './LandingPage';
import { MemoryRouter } from 'react-router';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <LandingPage />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
