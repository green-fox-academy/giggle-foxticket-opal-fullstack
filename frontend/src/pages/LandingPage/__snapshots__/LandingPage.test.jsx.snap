// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <section
    className="container"
  >
    <div
      className="left-half"
    >
      <article>
        <h1>
          Super Size Me
        </h1>
        <p>
          Director Morgan Spurlock's social experiment in fast-food gastronomy sees him attempting to subsist uniquely on food from the McDonald's menu for an entire month. In the process his weight balloons, his energy level plummets and he experiences all sorts of unexpected -- and terrifying -- side effects. He also examines the corporate giant's growing role in the lives of American consumers and explores its methods of indoctrinating young people and its contribution to America's obesity epidemic.
        </p>
      </article>
      <button
        className="btn btn--primary--solid btn--medium"
      >
        Learn More
      </button>
    </div>
    <div
      className="right-half"
    >
      <article>
        <img
          alt="hello"
          className="cat"
          src="card-cat.jpeg"
        />
      </article>
    </div>
  </section>,
  <div
    className="main-features"
  >
    <div
      className="card"
    >
      <div
        className="card-image"
      >
        <img
          alt=""
          src="card-cat.jpeg"
        />
      </div>
      <div
        className="card-copy"
      >
        <h1
          className="card-primary-header"
        >
          DOOM
        </h1>
        <h2
          className="card-secondary-header"
        >
          Video game series
        </h2>
        <p>
          Doom is a video game series and media franchise created by John Carmack, John Romero, Adrian Carmack, Kevin Cloud, and Tom Hall. The series focuses on the exploits of an unnamed space marine operating under the auspices of the Union Aerospace Corporation, who fights hordes of demons and the undead.
        </p>
      </div>
    </div>
    <div
      className="card"
    >
      <div
        className="card-image"
      >
        <img
          alt=""
          src="card-cat.jpeg"
        />
      </div>
      <div
        className="card-copy"
      >
        <h1
          className="card-primary-header"
        >
          Wolfenstein
        </h1>
        <h2
          className="card-secondary-header"
        >
          Video game series
        </h2>
        <p>
          Wolfenstein is a series of World War II video games. It was originally created by Muse Software, before being revived by id Software. The first two games in the series, Castle Wolfenstein and Beyond Castle Wolfenstein, were developed by Muse Software and focused on stealth-based gameplay from a top-down perspective.
        </p>
      </div>
    </div>
    <div
      className="card"
    >
      <div
        className="card-image"
      >
        <img
          alt=""
          src="card-cat.jpeg"
        />
      </div>
      <div
        className="card-copy"
      >
        <h1
          className="card-primary-header"
        >
          Quake
        </h1>
        <h2
          className="card-secondary-header"
        >
          Video game series
        </h2>
        <p>
          Quake is a series of first-person shooter video games, developed by id Software and, as of 2010, published by Bethesda Softworks. The series is composed of the eponymous game from 1996 and its nonlinear, standalone sequels which vary in setting and plot.
        </p>
      </div>
    </div>
  </div>,
  <a
    href="/shop"
    onClick={[Function]}
  >
    <button
      className="btn btn--danger--solid btn--medium"
    >
      Buy Tickets
    </button>
  </a>,
  <div
    className="video"
  >
    <div
      style={
        Object {
          "height": "360px",
          "width": "640px",
        }
      }
    >
      <div
        style={
          Object {
            "display": undefined,
            "height": "100%",
            "width": "100%",
          }
        }
      >
        <div />
      </div>
    </div>
  </div>,
  <form
    className="subscription-form"
    onSubmit={[Function]}
  >
    <label>
      Subscribe to our newsletter!
    </label>
    <div
      className="emailImg"
    >
      <img
        alt="subscribe email"
        src="default-mail.svg"
      />
    </div>
    <div
      className="form-group"
    >
      <input
        name="name"
        onChange={[Function]}
        placeholder="Your name"
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <input
        name="email"
        onChange={[Function]}
        placeholder="Your email"
        required={true}
        type="email"
        value=""
      />
      <p
        className=" isValid"
      >
        Username or Email is incorrect
        <svg
          color="red"
          fill="currentColor"
          height="1.5em"
          size="1.5em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": "red",
            }
          }
          viewBox="0 0 576 512"
          width="1.5em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
          />
        </svg>
      </p>
      <i
        className="fas fa-exclamation-triangle"
      />
    </div>
    <p
      className=" msg error"
      style={
        Object {
          "display": "none",
        }
      }
    >
      <span
        className=" errorSpan"
      >
          
      </span>
    </p>
    <button
      className="btn"
      type="submit"
    >
      Subscribe!
    </button>
  </form>,
  <footer>
    <div
      className="logos"
    >
      <img
        alt="Facebook"
        src="facebook.svg"
      />
      <img
        alt="Instagram"
        src="instagram.svg"
      />
      <img
        alt="Twitter"
        src="twitter.svg"
      />
    </div>
    <h3>
      I am the Footer
    </h3>
  </footer>,
]
`;
